truffle/src/main/java/org/jruby/truffle/RubyFiletypeDetector.java

```diff
@@ -0,0 +1,46 @@
+package org.jruby.truffle;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.spi.FileTypeDetector;
+import java.util.Optional;
+
+public final class RubyFiletypeDetector extends FileTypeDetector {
+
+    private final static String[] KNOWN_RUBY_FILES = new String[]{ "Gemfile", "Rakefile" };
+    private final static String[] KNOWN_RUBY_SUFFIXES = new String[]{ ".rb", ".rake", ".gemspec" };
+    private final static String RUBY_MIME = "application/x-ruby";
```

Comment:
What do you mean by `RubyLangauge`? I could not find a class like that, neither I found constant with `"application/x-ruby"`.